source/Player.hx

```diff
@@ -187,6 +210,48 @@ import flixel.group.FlxGroup;
     btmBox.y = y + height - hitBoxHeight;
   }
 
+  //Override player.kill to drop any weilded items
+  override public function kill():Void 
+  {
+	  if (weilding){
+		dropCurrentEquip();
+	  }
+	  super.kill();
+	  new FlxTimer().start(2, resetLevel, 1); //This helps speed things up for debugging
+  }
+  
+  private function resetLevel(Timer:FlxTimer):Void
```

Comment:
It felt more out of place in the Playstate, with the extra logical check since this is something that doesn't need one if it's called here in the Player. I can move the logic into the Playstate if necessary, but I still think it fits best here.